onnxruntime/core/providers/cpu/tensor/quantize_linear.cc

```diff
@@ -37,7 +45,15 @@ Status DequantizeLinear<T>::Compute(OpKernelContext* ctx) const {
   int64_t broadcast_dim;
   int64_t block_size;
 
-  if (has_axis_) {  // custom DequantizeLinear only
+  if (IsScalarOr1ElementVector(&x_scale)) {  // per-tensor DequantizeLinear
```

Comment:
It would be nice to either move this logic off to a PrepareForCompute style helper or register a single kernel for uint8/int8 to reduce code size. The ORT_ENFORCE code ends up fairly large.